google-cloud-pubsub/lib/google/cloud/pubsub/subscriber.rb

```diff
@@ -335,6 +337,21 @@ def inspect
 
         protected
 
+        def coerce_inventory inventory
+          @inventory = inventory
+          if @inventory.is_a? Hash
+            # Support deprecated field names
+            @inventory[:max_outstanding_messages] = @inventory[:limit] if @inventory[:limit]
+            @inventory[:max_outstanding_bytes] = @inventory[:bytesize] if @inventory[:bytesize]
+            @inventory[:max_total_lease_duration] = @inventory[:extension] if @inventory[:extension]
```

Comment:
Is the inventory hash visible outside the object? (Is it returned from an accessor anywhere?) If not, then we should just remove the old keys (`:limit` etc) during `coerce_inventory`. If it is visible, then we should ensure that the values of the deprecated keys remain identical to the new keys (e.g. currently the new keys are coerced to `Integer` whereas the old ones aren't)

Reply:
e.g.
```ruby
@inventory[:max_outstanding_messages] ||= @inventory.delete :limit
```

Also, we might want to dup the inventory hash, so that someone doesn't pass us a hash and then modify it out from under us later (which will be very confusing since we've now made modifications ourselves)